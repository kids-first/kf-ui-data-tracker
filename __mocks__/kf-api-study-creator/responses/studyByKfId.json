{
  "data": {
    "studyByKfId": {
      "id": "U3R1ZHlOb2RlOlNEXzhXWDhRUTA2",
      "name": "benchmark extensible e-business",
      "shortName": null,
      "bucket": "stay-wrong-ready",
      "kfId": "SD_8WX8QQ06",
      "createdAt": "2019-04-20T15:22:18.780221+00:00",
      "modifiedAt": "2019-04-25T15:22:18.780221+00:00",
      "attribution": null,
      "dataAccessAuthority": "dbGaP",
      "externalId": "test_study_external_id",
      "releaseStatus": null,
      "version": null,
      "releaseDate": "2019-08-24",
      "description": "Sudy description in markdown, commonly the X01 abstract text. --  As test study description",
      "anticipatedSamples": 100,
      "awardeeOrganization": "CHOP",
      "collaborators": {
        "edges": [
          {
            "node": {
              "id": "VXNlck5vZGU6NQ==",
              "username": "Jennifer Warren",
              "firstName": "Marc",
              "lastName": "Wilson",
              "email": "ericksontracy@benitez-ruiz.com",
              "picture": "",
              "roles": [],
              "__typename": "UserNode"
            },
            "__typename": "UserNodeEdge"
          },
          {
            "node": {
              "id": "VXNlck5vZGU6OQ==",
              "username": "Anthony Hamilton",
              "firstName": "John",
              "lastName": "Garza",
              "email": "richard06@merritt.net",
              "picture": "",
              "roles": [],
              "__typename": "UserNode"
            },
            "__typename": "UserNodeEdge"
          },
          {
            "node": {
              "id": "VXNlck5vZGU6MjY=",
              "username": "devadmin",
              "firstName": "",
              "lastName": "",
              "email": "admin@myproject.com",
              "picture": "",
              "roles": ["ADMIN"],
              "__typename": "UserNode"
            },
            "__typename": "UserNodeEdge"
          }
        ],
        "__typename": "UserNodeConnection"
      },
      "files": {
        "edges": [
          {
            "node": {
              "id": "RmlsZU5vZGU6U0ZfNVpQRU0xNjc=",
              "kfId": "SF_5ZPEM167",
              "name": "organization.jpeg",
              "fileType": "CLN",
              "tags": [],
              "description": "Month necessary animal end standard case. View system operation message decade. Actually sing because deal everything woman subject.",
              "downloadUrl": "http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167",
              "versions": {
                "edges": [
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl80TlpOQkVPRg==",
                      "kfId": "FV_NVV741M1",
                      "state": "PEN",
                      "size": 9893,
                      "createdAt": "2018-08-21T03:16:11+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl83NkhONTdaNw==",
                      "kfId": "FV_KCDKCDYR",
                      "state": "PEN",
                      "size": 5460,
                      "createdAt": "2018-09-27T05:32:26+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9IUkFHWE5XMQ==",
                      "kfId": "FV_800JI8IP",
                      "state": "PEN",
                      "size": 5419,
                      "createdAt": "2017-08-25T00:05:30+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  }
                ],
                "__typename": "VersionNodeConnection"
              },
              "__typename": "FileNode"
            },
            "__typename": "FileNodeEdge"
          },
          {
            "node": {
              "id": "RmlsZU5vZGU6U0ZfWTA3SU4xSE8=",
              "kfId": "SF_Y07IN1HO",
              "name": "its.mp3",
              "fileType": "SHM",
              "tags": ["dbGaP", "random", "tag that has ten more char"],
              "description": "Question meeting move recognize.",
              "downloadUrl": "http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_Y07IN1HO",
              "versions": {
                "edges": [
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9UWjVROEhYWA==",
                      "kfId": "FV_IJ2CN4P9",
                      "state": "PEN",
                      "size": 7394,
                      "createdAt": "2017-08-22T08:32:38+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9aOUM1SUUwRg==",
                      "kfId": "FV_DNTN03CS",
                      "state": "PEN",
                      "size": 5883,
                      "createdAt": "2017-09-02T21:02:35+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9ITDg0SDhHTw==",
                      "kfId": "FV_ZDB8JOCT",
                      "state": "PEN",
                      "size": 1899,
                      "createdAt": "2019-03-14T05:46:35+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9YRVg4TERHMw==",
                      "kfId": "FV_10SBI6VA",
                      "state": "PEN",
                      "size": 7631,
                      "createdAt": "2018-03-13T23:52:08+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9HV1JLUUVDQQ==",
                      "kfId": "FV_8KF1VMN9",
                      "state": "PEN",
                      "size": 6834,
                      "createdAt": "2019-04-06T07:51:17+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  }
                ],
                "__typename": "VersionNodeConnection"
              },
              "__typename": "FileNode"
            },
            "__typename": "FileNodeEdge"
          }
        ],
        "__typename": "FileNodeConnection"
      },
      "events": {
        "edges": [
          {
            "node": {
              "id": "RXZlbnROb2RlOjM5",
              "uuid": "b790803e-580e-458c-8b42-78252a5403f6",
              "createdAt": "2019-08-26T18:25:02.985155+00:00",
              "eventType": "FV_CRE",
              "description": "devadmin created version FV_9AF7CC3Z of file SF_5ZPEM167",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjM4",
              "uuid": "6372f6f7-8556-4b1c-9c29-7c0a7923f7ca",
              "createdAt": "2019-08-26T18:24:46.341665+00:00",
              "eventType": "FV_UPD",
              "description": "devadmin updated version FV_4FCEYZ3T of file SF_5ZPEM167",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjM2",
              "uuid": "ee949b92-ddb0-491b-b807-8f754e1641ff",
              "createdAt": "2019-08-26T18:24:36.763314+00:00",
              "eventType": "FV_CRE",
              "description": "devadmin created version FV_ABCPEQRK of file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjM1",
              "uuid": "05379686-3d68-4fed-b6a5-a5e47d97996f",
              "createdAt": "2019-08-26T18:24:19.986588+00:00",
              "eventType": "FV_CRE",
              "description": "devadmin created version FV_BHXN2F7A of file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjM0",
              "uuid": "1ec8bb79-b6ad-47eb-bd40-7ffc81135abe",
              "createdAt": "2019-08-26T18:24:03.184513+00:00",
              "eventType": "FV_UPD",
              "description": "devadmin updated version FV_Z0NZQZNX of file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjMz",
              "uuid": "b9b6bf82-1bdc-4d79-b69b-5e0350d64e52",
              "createdAt": "2019-08-26T18:24:03.182570+00:00",
              "eventType": "SF_UPD",
              "description": "devadmin updated file SF_5ZPEM167",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjMy",
              "uuid": "02564a4f-ead8-4ebb-af12-7171dcc05fd4",
              "createdAt": "2019-08-26T18:23:53.703031+00:00",
              "eventType": "FV_CRE",
              "description": "devadmin created version FV_4FCEYZ3T of file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjMx",
              "uuid": "578c24b2-dcf3-4a80-a4e8-d240e9549736",
              "createdAt": "2019-08-26T18:23:53.695244+00:00",
              "eventType": "SF_CRE",
              "description": "devadmin created file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          }
        ],
        "__typename": "EventNodeConnection"
      },
      "projects": {
        "edges": [
          {
            "node": {
              "id": "UHJvamVjdE5vZGU6emh1eDMvc2QtNTUwNDB4bmY=",
              "createdBy": "zhux3",
              "createdOn": "2019-08-26T18:22:41+00:00",
              "description": "",
              "modifiedOn": "2019-08-26T18:22:41+00:00",
              "name": "test study name",
              "projectId": "zhux3/sd-8wx8qq06",
              "projectType": "DEL",
              "url": "https://cavatica-api.sbgenomics.com/v2/projects/zhux3/sd-8wx8qq06",
              "workflowType": null,
              "deleted": false,
              "__typename": "ProjectNode",
              "study": {
                "id": "U3R1ZHlOb2RlOlNEXzhXWDhRUTA2",
                "kfId": "SD_8WX8QQ06",
                "name": "benchmark extensible e-business",
                "shortName": null,
                "createdAt": "2019-04-20T15:22:18.780221+00:00",
                "modifiedAt": "2019-04-25T15:22:18.780221+00:00",
                "__typename": "StudyNode"
              }
            },
            "__typename": "ProjectNodeEdge"
          },
          {
            "node": {
              "id": "UHJvamVjdE5vZGU6emh1eDMvc2QtNTUwNDB4bmYtYndhLW1lbQ==",
              "createdBy": "zhux3",
              "createdOn": "2019-08-26T18:22:42+00:00",
              "description": "",
              "modifiedOn": "2019-08-26T18:22:42+00:00",
              "name": "test study name bwa-mem",
              "projectId": "zhux3/sd-8wx8qq06-bwa-mem",
              "projectType": "HAR",
              "url": "https://cavatica-api.sbgenomics.com/v2/projects/zhux3/sd-8wx8qq06-bwa-mem",
              "workflowType": "bwa_mem",
              "deleted": false,
              "__typename": "ProjectNode",
              "study": {
                "id": "U3R1ZHlOb2RlOlNEXzhXWDhRUTA2",
                "kfId": "SD_8WX8QQ06",
                "name": "benchmark extensible e-business",
                "shortName": null,
                "createdAt": "2019-04-20T15:22:18.780221+00:00",
                "modifiedAt": "2019-04-25T15:22:18.780221+00:00",
                "__typename": "StudyNode"
              }
            },
            "__typename": "ProjectNodeEdge"
          },
          {
            "node": {
              "id": "UHJvamVjdE5vZGU6emh1eDMvc2QtNTUwNDB4bmYtZ2F0ay1oYXBsb3R5cGVjYWxsZXI=",
              "createdBy": "zhux3",
              "createdOn": "2019-08-26T18:22:43+00:00",
              "description": "",
              "modifiedOn": "2019-08-26T18:22:43+00:00",
              "name": "test study name gatk-haplotypecaller",
              "projectId": "zhux3/sd-8wx8qq06-gatk-haplotypecaller",
              "projectType": "HAR",
              "url": "https://cavatica-api.sbgenomics.com/v2/projects/zhux3/sd-8wx8qq06-gatk-haplotypecaller",
              "workflowType": "gatk_haplotypecaller",
              "deleted": false,
              "__typename": "ProjectNode",
              "study": {
                "id": "U3R1ZHlOb2RlOlNEXzhXWDhRUTA2",
                "kfId": "SD_8WX8QQ06",
                "name": "benchmark extensible e-business",
                "shortName": null,
                "createdAt": "2019-04-20T15:22:18.780221+00:00",
                "modifiedAt": "2019-04-25T15:22:18.780221+00:00",
                "__typename": "StudyNode"
              }
            },
            "__typename": "ProjectNodeEdge"
          }
        ],
        "__typename": "ProjectNodeConnection"
      },
      "__typename": "StudyNode"
    }
  }
}
