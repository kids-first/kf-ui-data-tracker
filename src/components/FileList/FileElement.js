import React, {useState, useEffect} from 'react';
import PropTypes from 'prop-types';
import {withRouter, Link} from 'react-router-dom';
import TimeAgo from 'react-timeago';
import Badge from '../Badge/Badge';
import {Header, Table, Icon, List, Responsive} from 'semantic-ui-react';

import FileActionsContainer from '../../containers/FileActionsContainer';
import {
  fileSortedVersions,
  fileLatestDate,
  fileLatestSize,
  lengthLimit,
} from '../../common/fileUtils';
import {fileTypeDetail} from '../../common/enums';
import {longDate} from '../../common/dateUtils';

const useRecentlyUpdated = (latestDate, fileId) => {
  const [justUpdated, setJustUpdated] = useState(false);

  useEffect(() => {
    let FIVE_MIN = 300000;
    let docs = sessionStorage.getItem('kf_updated_docs');
    // make sure we only select one row
    if (fileId === docs) {
      // make sure that row was updated less than 5 minutes ago
      if (new Date().getTime() - Date.parse(latestDate) < FIVE_MIN) {
        setJustUpdated(true);
        setTimeout(() => {
          // reset after 7seconds
          setJustUpdated(false);
          sessionStorage.removeItem('kf_updated_docs');
        }, 7000);
      }
    }
    return () => {
      setJustUpdated(false);
    };
  }, [latestDate, fileId]);

  return justUpdated;
};

/**
 * Displays a list of file attributes
 */
const FileAttributes = ({
  latestDate,
  fileSize,
  fileType,
  horizontal,
  fileVersions,
  fileKfID,
}) => {
  const justUpdated = useRecentlyUpdated(latestDate, fileKfID);

  return (
    <List horizontal={horizontal} link={!justUpdated}>
      <List.Item>
        <List.Content verticalAlign="middle">
          <Icon
            name={`${fileType.icon || 'question'}`}
            size="small"
            color="grey"
          />
          {fileType.title}
        </List.Content>
      </List.Item>

      <List.Item>
        <List.Content>
          {justUpdated ? <Icon name="refresh" /> : null}
          {latestDate ? (
            <>
              Modified{' '}
              <TimeAgo
                date={latestDate}
                live={false}
                title={longDate(latestDate)}
              />
            </>
          ) : (
            'Unknown time'
          )}
        </List.Content>
      </List.Item>
      <List.Item>
        <List.Content>{fileVersions} versions </List.Content>
      </List.Item>
      <List.Item>
        <List.Content>{fileSize}</List.Content>
      </List.Item>
    </List>
  );
};

FileAttributes.propTypes = {
  /** The kf id of the file */
  fileKfID: PropTypes.string.isRequired,
  /** The date of last modification */
  latestDate: PropTypes.string.isRequired,
  /** The size of the file */
  fileSize: PropTypes.string.isRequired,
  /** The type of file */
  fileType: PropTypes.object.isRequired,
  /** Whether the list should be displayed horizontally */
  horizontal: PropTypes.bool,
};

FileAttributes.defaultProps = {
  horizontal: false,
};

/**
 * Displays unordered study files in list view
 */
const FileElement = ({fileNode, loading, history, match, fileListId}) => {
  const fileKfID = fileNode.kfId || 'unknown ID';
  const fileName = fileNode.name || 'unknown file name';
  const fileDescription = fileNode.description || null;
  const sortedVersions = fileSortedVersions(fileNode);
  const latestDate = fileLatestDate(sortedVersions);
  const fileSize = fileLatestSize(sortedVersions);
  const fileType =
    fileNode && fileTypeDetail[fileNode.fileType]
      ? fileTypeDetail[fileNode.fileType]
      : {title: 'unknown', icon: 'question'};
  const versionState =
    sortedVersions.length > 0 ? sortedVersions[0].node.state : null;

  const justUpdated = useRecentlyUpdated(latestDate, fileKfID);
  return (
    <Table.Row
      style={{backgroundColor: justUpdated ? '#f8ffff' : 'inherit'}}
      data-testid="file-item"
      className="cursor-pointer"
      onClick={() =>
        history.push(`/study/${match.params.kfId}/documents/${fileKfID}`)
      }
    >
      <Table.Cell textAlign="center">
        <Header>
          {justUpdated && (
            <>
              <Badge state="UPD" icon="refresh" className="mb-5" />
              <br />
            </>
          )}

          <Badge
            state={versionState}
            loading={loading}
            filled={versionState === 'CHN'}
          />
        </Header>
      </Table.Cell>
      <Table.Cell>
        <Header size="medium" as="span">
          <Link to={`/study/${match.params.kfId}/documents/${fileKfID}`}>
            {fileName}
          </Link>
        </Header>
        <p className="noMargin">
          {fileDescription ? <>{lengthLimit(fileDescription, 100)}</> : null}
        </p>
        <Responsive
          as={FileAttributes}
          minWidth={Responsive.onlyTablet.minWidth}
          fileKfID={fileKfID}
          latestDate={latestDate}
          fileSize={fileSize}
          fileType={fileType}
          fileVersions={fileNode.versions.edges.length}
          horizontal={true}
        />
        <Responsive
          as={FileAttributes}
          maxWidth={Responsive.onlyTablet.minWidth}
          fileKfID={fileKfID}
          latestDate={latestDate}
          fileSize={fileSize}
          fileType={fileType}
          fileVersions={fileNode.versions.edges.length}
          horizontal={false}
        />
      </Table.Cell>
      <Table.Cell textAlign="center">
        <Responsive
          as={FileActionsContainer}
          minWidth={Responsive.onlyTablet.minWidth}
          node={fileNode}
          studyId={fileListId}
          vertical={true}
          fluid={false}
        />
        <Responsive
          as={FileActionsContainer}
          maxWidth={Responsive.onlyTablet.minWidth}
          node={fileNode}
          studyId={fileListId}
          vertical={false}
          fluid={true}
        />
      </Table.Cell>
    </Table.Row>
  );
};

FileList.propTypes = {
  /** Array of study object*/
  fileNode: PropTypes.object,
};

FileList.defaultProps = {
  fileNode: null,
};

export default withRouter(FileElement);
