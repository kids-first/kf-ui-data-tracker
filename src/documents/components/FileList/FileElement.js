import React, {useState, useEffect} from 'react';
import PropTypes from 'prop-types';
import {withRouter, Link} from 'react-router-dom';
import TimeAgo from 'react-timeago';
import {Table, Icon, Checkbox, Popup} from 'semantic-ui-react';
import Markdown from 'react-markdown';
import FileActionButtons from '../FileActionButtons/FileActionButtons';
import {fileSortedVersions, fileLatestDate} from '../../utilities';
import {fileTypeDetail} from '../../../common/enums';
import {longDate} from '../../../common/dateUtils';
import FileTags from '../FileDetail/FileTags';

const useRecentlyUpdated = (latestDate, fileId) => {
  const [justUpdated, setJustUpdated] = useState(false);

  useEffect(() => {
    let FIVE_MIN = 300000;
    let docs = sessionStorage.getItem('kf_updated_docs');
    // make sure we only select one row
    if (fileId === docs) {
      // make sure that row was updated less than 5 minutes ago
      if (new Date().getTime() - Date.parse(latestDate) < FIVE_MIN) {
        setJustUpdated(true);
        setTimeout(() => {
          // reset after 7seconds
          setJustUpdated(false);
          sessionStorage.removeItem('kf_updated_docs');
        }, 7000);
      }
    }
    return () => {
      setJustUpdated(false);
    };
  }, [latestDate, fileId]);

  return justUpdated;
};

/**
 * Displays unordered study files in list view
 */
const FileElement = ({
  fileNode,
  loading,
  history,
  match,
  fileListId,
  updateFile,
  deleteFile,
  downloadFileMutation,
  selected,
  onSelectOne,
}) => {
  const fileKfID = fileNode.kfId || 'unknown ID';
  const fileName = fileNode.name || 'unknown file name';
  // Show at most 5 lines of the description
  const fileDescription =
    fileNode.description
      .split('\n')
      .splice(0, 5)
      .join('\n') || '';
  const sortedVersions = fileSortedVersions(fileNode);
  const latestDate = fileLatestDate(sortedVersions);
  const fileType =
    fileNode && fileTypeDetail[fileNode.fileType]
      ? fileTypeDetail[fileNode.fileType]
      : {title: 'unknown', icon: 'question'};
  const justUpdated = useRecentlyUpdated(latestDate, fileKfID);
  return (
    <Table.Row
      style={{backgroundColor: justUpdated ? '#f8ffff' : 'inherit'}}
      data-testid="file-item"
      className="cursor-pointer"
      onClick={() =>
        history.push(`/study/${match.params.kfId}/documents/${fileKfID}`)
      }
    >
      <Table.Cell
        textAlign="center"
        onClick={e => {
          e.stopPropagation();
          onSelectOne(fileKfID);
        }}
      >
        <Checkbox data-testid="file-select" checked={selected} />
      </Table.Cell>
      <Popup
        wide="very"
        mouseEnterDelay={500}
        header={fileType.title || 'Unknown Type'}
        content={fileType.description || 'Unknown Type'}
        trigger={
          <Table.Cell textAlign="center">
            <Icon name={fileType.icon || 'question'} />
          </Table.Cell>
        }
      />
      <Table.Cell>
        {fileDescription.length > 3 && (
          <Popup
            offset="-75%"
            wide="very"
            position="top left"
            trigger={<Icon name="info circle" />}
            content={
              <Markdown
                source={fileDescription}
                renderers={{
                  image: Image,
                  table: props => <Table>{props.children}</Table>,
                }}
              />
            }
          />
        )}
        <Link to={`/study/${match.params.kfId}/documents/${fileKfID}`}>
          {fileName}
        </Link>
      </Table.Cell>
      <Table.Cell textAlign="center" width="4">
        <FileTags fileNode={fileNode} updateFile={updateFile} />
      </Table.Cell>
      <Table.Cell textAlign="center" width="1">
        <TimeAgo date={latestDate} live={false} title={longDate(latestDate)} />
      </Table.Cell>
      <Table.Cell textAlign="center">
        <FileActionButtons
          fluid
          node={fileNode}
          studyId={fileListId}
          deleteFile={deleteFile}
          downloadFileMutation={downloadFileMutation}
        />
      </Table.Cell>
    </Table.Row>
  );
};

FileList.propTypes = {
  /** Array of study object*/
  fileNode: PropTypes.object,
};

FileList.defaultProps = {
  fileNode: null,
};

export default withRouter(FileElement);
