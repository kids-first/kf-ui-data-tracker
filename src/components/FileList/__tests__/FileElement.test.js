import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {render} from 'react-testing-library';
import FileElement from '../FileElement';
import studyByKfId from './studyByKfId';

it('renders correctly', () => {
  const file = studyByKfId.data.studyByKfId.files.edges[0].node;
  const tree = render(
    <MemoryRouter>
      <FileElement fileNode={file} />
    </MemoryRouter>,
  );
  expect(tree.container).toMatchSnapshot();
});

it('renders loading state', () => {
  const tree = render(
    <MemoryRouter>
      <FileElement loading={true} />
    </MemoryRouter>,
  );
  expect(tree.container).toMatchSnapshot();
});

it('renders error state', () => {
  const tree = render(
    <MemoryRouter>
      <FileElement
        error={{graphQLErrors: [{message: 'something went wrong'}]}}
      />
    </MemoryRouter>,
  );
  expect(tree.container).toMatchSnapshot();
  expect(tree.queryByText('something went wrong')).not.toBeNull();
});
