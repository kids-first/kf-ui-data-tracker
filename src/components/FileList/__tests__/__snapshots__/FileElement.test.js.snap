// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
        style="background-color: inherit;"
      >
        <td
          class="center aligned"
        >
          <div
            class="ui header"
          >
            <div
              class="ui orange tiny basic label"
            >
              Pending review
            </div>
          </div>
        </td>
        <td
          class=""
        >
          <span
            class="ui medium header"
          >
            <a
              href="/study/undefined/documents/SF_5ZPEM167"
            >
              organization.jpeg
            </a>
          </span>
          <p
            class="noMargin"
          >
            Month necessary animal end standard case. View system operation message decade. Actually sing becaus...
          </p>
          <div
            class="ui horizontal link list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              <div
                class="middle aligned content"
              >
                <i
                  aria-hidden="true"
                  class="grey hospital small icon"
                />
                Clinical/Phenotype Data
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="content"
              >
                Modified
                 
                <time
                  datetime="2018-09-27T05:32:26.000Z"
                  title="27 Sept 2018"
                >
                  7 months ago
                </time>
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="content"
              >
                3
                 version
                s
                 
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="content"
              >
                5.5 kB
              </div>
            </div>
          </div>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui small buttons"
          >
            <button
              class="ui basic compact icon button"
            >
              <i
                aria-hidden="true"
                class="copy icon"
              />
            </button>
            <button
              class="ui basic compact icon button"
              data-testid="download-file"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders latest temporary state 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
        style="background-color: inherit;"
      >
        <td
          class="center aligned"
        >
          <div
            class="ui header"
          >
            <div
              class="ui orange tiny basic label"
            >
              Pending review
            </div>
          </div>
        </td>
        <td
          class=""
        >
          <span
            class="ui medium header"
          >
            <a
              href="/study/undefined/documents/SF_5ZPEM167"
            >
              organization.jpeg
            </a>
          </span>
          <p
            class="noMargin"
          >
            Month necessary animal end standard case. View system operation message decade. Actually sing becaus...
          </p>
          <div
            class="ui horizontal link list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              <div
                class="middle aligned content"
              >
                <i
                  aria-hidden="true"
                  class="grey hospital small icon"
                />
                Clinical/Phenotype Data
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="content"
              >
                Modified
                 
                <time
                  datetime="2018-09-27T05:32:27.000Z"
                  title="27 Sept 2018"
                >
                  7 months ago
                </time>
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="content"
              >
                3
                 version
                s
                 
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="content"
              >
                5.5 kB
              </div>
            </div>
          </div>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui small buttons"
          >
            <button
              class="ui basic compact icon button"
            >
              <i
                aria-hidden="true"
                class="copy icon"
              />
            </button>
            <button
              class="ui basic compact icon button"
              data-testid="download-file"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders loading state 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
        style="background-color: inherit;"
      >
        <td
          class="center aligned"
        >
          <div
            class="ui header"
          >
            <div
              class="ui orange tiny basic label"
            >
              Pending review
            </div>
          </div>
        </td>
        <td
          class=""
        >
          <span
            class="ui medium header"
          >
            <a
              href="/study/undefined/documents/SF_5ZPEM167"
            >
              organization.jpeg
            </a>
          </span>
          <p
            class="noMargin"
          >
            Month necessary animal end standard case. View system operation message decade. Actually sing becaus...
          </p>
          <div
            class="ui horizontal link list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              <div
                class="middle aligned content"
              >
                <i
                  aria-hidden="true"
                  class="grey hospital small icon"
                />
                Clinical/Phenotype Data
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="content"
              >
                Modified
                 
                <time
                  datetime="2018-09-27T05:32:26.000Z"
                  title="27 Sept 2018"
                >
                  7 months ago
                </time>
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="content"
              >
                3
                 version
                s
                 
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="content"
              >
                5.5 kB
              </div>
            </div>
          </div>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui small buttons"
          >
            <button
              class="ui basic compact icon button"
            >
              <i
                aria-hidden="true"
                class="copy icon"
              />
            </button>
            <button
              class="ui basic compact icon button"
              data-testid="download-file"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
